js/cards.js

```diff
@@ -0,0 +1,112 @@
+import {createAdds} from './data.js';
+
+const cardPlace = document.querySelector('.map');
+const cardPlaceElement = cardPlace.querySelector('#map-canvas');
+
+// Находим в содержимом шаблона блок с классом popup
+const cardTemplate = document.querySelector('#card').content.querySelector('.popup');
+
+// Присваиваем переменной значение функции по созданию массива с объектами
+const similarAdds = createAdds();
+
+// Словарь видов жилья
+const accommodationType = {
+  flat: 'Квартира',
+  bungalow: 'Бунгало',
+  house: 'Дом',
+  palace: 'Дворец',
+  hotel: 'Отель'
+};
+
+const generateFeatures = (featureArray) => {
+
+  // Создаём документ-фрагмент
+  const featuresFragmentElement = document.createDocumentFragment();
+  // Клонируем список
+  const featureTemplateElement = cardTemplate.querySelector('.popup__features').cloneNode(true);
+  // Создаём коллекцию (массив) элементов списка
+  const featureListTemplateElement = featureTemplateElement.querySelectorAll('.popup__feature');
+  // Создаём функцию по созданию модифицированного массива
+  const makeFeatureArray = featureArray.map((item) => `popup__feature--${item}`);
+
+  // Для каждого элемента списка проводим сл.операцию
+  featureListTemplateElement.forEach((listItem) => {
+    // С помощью свойства classList выбираем эл-т popup__feature--*** и присваиваем его значение переменной
+    const cssModifier = listItem.classList[1];
+    // Если в модифицированном массиве есть модификатор, то
+    if (makeFeatureArray.includes(cssModifier)) {
+      // добавляем элемент списка в документ-фрагмент
+      featuresFragmentElement.appendChild(listItem);
+    }
+  });
+
+  return featuresFragmentElement;
+};
+
+const generatePhotos = (photoArray) => {
+  const photoFragmentElement = document.createDocumentFragment();
+
+  for (let i = 0; i < photoArray.length; i++) {
+    const photoListTemplateElement = cardTemplate.querySelector('.popup__photo').cloneNode(true);
+    photoListTemplateElement.src = photoArray[i];
+    photoFragmentElement.appendChild(photoListTemplateElement);
+  }
+
+  return photoFragmentElement;
+};
+
+// Функция по добавлению в объявление типа жилья
+const createAccommodationType = (accommodationName, addTemplateElement) => {
```

Comment:
```suggestion
const getAccommodationType = (type, element) => {
```

`get` - получить. Нам надо _получить_ тип жилья, а не создать тип жилья, потому что _создание_ - это что-то новое, а нам надо из уже существующих типов получить нужный нам. Поэтому get